src/errors.js

```diff
@@ -1,896 +1,81 @@
-  constructor(m) {   
-    super(m);
-  }
-}
-
-class QuerySavedInvalidCachePeriodValueError extends SlicingDiceError {
-  constructor(m) {   
-    super(m);
-  }
-}
-
-class QuerySavedInvalidNameError extends SlicingDiceError {
-  constructor(m) {   
-    super(m);
-  }
-}
-
-class QueryCountInvalidParameterError extends SlicingDiceError {
-  constructor(m) {   
-    super(m);
-  }
-}
-
-class QueryAggregationInvalidParameterError extends SlicingDiceError {
-  constructor(m) {   
-    super(m);
-  }
-}
-
-class QueryAggregationInvalidFilterQueryError extends SlicingDiceError {
-  constructor(m) {   
-    super(m);
-  }
-}
-
-class QueryInvalidMinfreqError extends SlicingDiceError {
-    constructor(m) {
-        super(m);
+  constructor(data) {
+    let message = '';
+    if (data.code) {
+      message += 'code=' + data.code + ' ';
     }
-}
-
-class QueryExceededMaxNumberQuerysError extends SlicingDiceError {
-    constructor(m) {
-        super(m);
+    if (data.message) {
+      message += 'message="' + data.message + '" ';
     }
-}
-
-class QueryInvalidOperatorUsageError extends SlicingDiceError {
-    constructor(m) {
-        super(m);
+    if (data.more_info) {
+      message += 'more_info="' + JSON.stringify(data.more_info) + '" ';
     }
-}
 
-class QueryInvalidParameterUsageError extends SlicingDiceError {
-    constructor(m) {
-        super(m);
-    }
-}
+    super(message);
 
-class QueryParameterInvalidColumnUsageError extends SlicingDiceError {
-    constructor(m) {
-        super(m);
-    }
-}
+    this.code = data.code;
+    this.more_info = data.more_info; 
 
-class QueryInvalidColumnUsageError extends SlicingDiceError {
-    constructor(m) {
-        super(m);
+    if (typeof Error.captureStackTrace === 'function') {
+      Error.captureStackTrace(this, this.constructor);
+    } else { 
+      this.stack = (new Error(data.message)).stack; 
     }
-}
-
-
-class ColumnCreateInternalError extends ExtendableError {
-  constructor(m) {   
-    super(m);
-  }
-}
-
-// CLIENT ERRORS
-class InternalError extends ExtendableError {
-  constructor(m) {   
-    super(m);
-  }
-}
-
-class InvalidQueryTypeError extends ExtendableError {
-  constructor(m) {   
-    super(m);
-  }
-}
-
-class MaxLimitError extends ExtendableError {
-  constructor(m) {   
-    super(m);
   }
 }
 
-class InvalidQueryError extends ExtendableError {
-  constructor(m) {   
-    super(m);
-  }
-}
-
-class InvalidColumnDescriptionError extends ExtendableError {
-  constructor(m) {   
-    super(m);
+class SlicingDiceError extends ExtendableError {
+  constructor(data) {   
+    super(data);
   }
 }
 
 class SlicingDiceClientError extends ExtendableError {
-  constructor(m) {   
-    super(m);
+  constructor(data) {   
+    super(data);
   }
 }
 
+// Specific Errors
 
-class WrongTypeError extends ExtendableError {
-  constructor(m) {   
-    super(m);
+class DemoUnavailableException extends SlicingDiceError {
```

Comment:
Should it be "DemoUnavailableException" or "DemoUnavailableError"?
This also applies to other modifications.